src/Planner/PlannerExpressionAnalysis.cpp

```diff
@@ -64,6 +64,10 @@ bool isDeterministicConstant(const ConstantNode & root)
         const auto * node = nodes.top();
         nodes.pop();
 
+        if (node->getNodeType() == QueryTreeNodeType::QUERY)
+            /// Allow scalar subqueries and IN; we send them to all the shards.
```

Comment:
Sorry for my lack of knowledge about the context: Does this mean that we send the result of the scalar subquery / result of IN, or that we send the subquery themselves?

And another thing, but it depends on the previous response: would it make sense to change the name of the function (if we send the subquery itself)? At this point I only know it has to do with distributed queries because I've seen other related changes so maybe it could be more clear or we could add some comments to the function itself to explain what is used for? Thanks!

Reply:
I also don't understand this change. It sounds strange that we treat scalar subquery as a **deterministic** constant.

Reply:
This function is used to check whether the constant values is the same for all the replicas/shards. If the constant was calculated from the result of scalar subquery, or the constant was calculated as `const_expr IN set`, it would be the same : we send the result of scalar subqueries in the protocol.

Please, opt for the better function name.